tests/tests_fabric/strategies/test_xla.py

```diff
@@ -148,3 +151,32 @@ def tpu_all_gather_fn(strategy):
 def test_tpu_all_gather():
     """Test the all_gather operation on TPU."""
     xla_launch(tpu_all_gather_fn)
+
+
+def tpu_broadcast_master_params_fn(broadcast_master_params, strategy):
+    seed_everything(random.seed())
+    model = torch.nn.Linear(1, 1).to(strategy.root_device)
+    model = strategy.setup_module(model)
+    gathered = strategy.all_gather(model.weight)
+    if broadcast_master_params:
+        for t in gathered:
+            assert gathered[0] == t
+    else:
+        with pytest.raises(AssertionError):
+            for t in gathered:
+                assert gathered[0] == t
+
+
+@RunIf(tpu=True)
+@pytest.mark.parametrize("broadcast_master_params", [True, False])
+@mock.patch.dict(os.environ, {"PJRT_DEVICE": "TPU"}, clear=True)
```

Comment:
This isn't necessary. Our CI sets it: https://github.com/Lightning-AI/lightning/actions/runs/4877179869/jobs/8701590389?pr=17522#step:7:66

```suggestion
```